tensorflow/core/kernels/dynamic_partition_op_gpu.cu.cc

```diff
@@ -413,9 +420,13 @@ class DynamicPartitionOpGPU : public AsyncOpKernel {
     // Determine temporary device storage requirements
     Tensor cub_temp_storage;
     size_t temp_storage_bytes = 0;
-    gpuprim::DeviceReduce::ReduceByKey(
-        NULL, temp_storage_bytes, keys_in_ptr, unique_out_it, values_in,
-        aggregates_out_it, num_runs_ptr, reduction_op, N, cu_stream);
+    auto gpuResult = gpuprim::DeviceReduce::ReduceByKey(
+              NULL, temp_storage_bytes, keys_in_ptr, unique_out_it, values_in,
+              aggregates_out_it, num_runs_ptr, reduction_op, N, cu_stream);
+    
+    OP_REQUIRES(c, gpuResult == gpuSuccess, 
+        errors::Internal("Failed to launch copy from device to host."));
```

Comment:
errors::Internal("Failed to launch gpuprim::DeviceRadixSort::SortPairs to calculate temp_storage_bytes, status: ", cudaGetErrorString(gpuResult));